src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,180 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private T item;
+        private Node<T> next;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        indexValidation(index, size);
+        if (index == 0) {
+            addFirst(value);
+        } else if (index == size) {
+            addLast(value);
+        } else {
+            Node<T> oldNode = findElenemt(index);
+            Node<T> newNode = new Node<T>(oldNode.prev, value, oldNode);
+            oldNode.prev.next = newNode;
+            oldNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index, size - 1);
+        return findElenemt(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T removed = remove(index);
+        add(value, index);
+        return removed;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidation(index, size - 1);
+        if (index == 0) {
+            return removeFirst();
+        } else if (index == size - 1) {
+            return removeLast();
+        } else {
+            Node<T> oldNode = findElenemt(index);
+            unlink(oldNode);
+            size--;
+            return oldNode.item;
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        int index = -1;
+        for (int i = 0; i < size; i++) {
+            T element = get(i);
+            if (element == object || (element != null && element.equals(object))) {
+                index = i;
+                break;
+            }
+        }
```

Comment:
In each iteration, you are searching for an element by index and checking the index in the method `get`
Now the complexity is O(n^2). Let's think about how you can optimize your algorithm

Reply:
Oh, such a tricky downside of reusing the methods! Well, in the new commit I tried to make it O(n), although now it looks a bit longer.